color/src/parse.rs

```diff
@@ -461,31 +461,53 @@ impl<'a> Parser<'a> {
 /// Tries to return a suitable error for any invalid string, but may be
 /// a little lax on some details.
 pub fn parse_color_prefix(s: &str) -> Result<(usize, DynamicColor), ParseError> {
+    #[inline]
+    fn set_from_named_color_space(mut color: DynamicColor) -> DynamicColor {
+        color.flags.set_named_color_space();
+        color
+    }
+
     if let Some(stripped) = s.strip_prefix('#') {
         let (ix, channels) = get_4bit_hex_channels(stripped)?;
         let color = color_from_4bit_hex(channels);
-        return Ok((ix + 1, DynamicColor::from_alpha_color(color)));
+        // Hex colors are seen as if they are generated from the named `rgb()` color space
+        // function.
+        let mut color = DynamicColor::from_alpha_color(color);
+        color.flags.set_named_color_space();
+        return Ok((ix + 1, color));
     }
     let mut parser = Parser::new(s);
     if let Some(id) = parser.ident() {
         let color = match id {
-            "rgb" | "rgba" => parser.rgb(),
-            "lab" => parser.lab(100.0, 1.25, ColorSpaceTag::Lab),
-            "lch" => parser.lch(100.0, 1.25, ColorSpaceTag::Lch),
-            "oklab" => parser.lab(1.0, 0.004, ColorSpaceTag::Oklab),
-            "oklch" => parser.lch(1.0, 0.004, ColorSpaceTag::Oklch),
-            "hsl" | "hsla" => parser.hsl(),
-            "hwb" => parser.hwb(),
+            "rgb" | "rgba" => parser.rgb().map(set_from_named_color_space),
```

Comment:
Not a huge fan of this pattern, but I'm not sure how to avoid it...

Reply:
You mean the `map(set_from_named_color_space)`? I agree; I remember I tried some alternatives and that this pattern was the least bad.